framework/src/problems/FEProblemBase.C

```diff
@@ -824,6 +824,7 @@ FEProblemBase::initialSetup()
       reinitScalars(tid);
 
     execute(EXEC_INITIAL);
+    setCurrentExecuteOnFlag(EXEC_INITIAL);
```

Comment:
Just checking, should this be before the `execute` line?

Reply:
This is needed, I will add a comment regarding the reason.